.github/workflows/python-publish.yml

```diff
@@ -17,11 +17,11 @@ jobs:
     - name: Set up Python
       uses: actions/setup-python@v2
```

Comment:
while you are at it, you can set the base action to their most up to date: v4:

line 16:     - uses: actions/checkout@v4
line 18:       uses: actions/setup-python@v2

Reply:
sorry  I messed it up: use v4 for both lines!

Reply:
A GitHub actions version updater ([see here](https://github.com/marketplace/actions/github-actions-version-updater)) gave me this:

- uses: actions/checkout@v4.2.0
- uses: actions/setup-python@v5.2.0

Do I use these or do I just put v4 ?

Reply:
Yes it is possible, you will only need to ensure that the repository secret variable `WORKFLOW_SECRET` is defined with a token with **workflow** scope enabled.

Reply:
ok i did it, could you try?

Reply:
I did some tests in my dev branch of the repository and everything works fine.

When the **GitHub Actions Version Updater** is triggered it automatically creates a pull request to update versions in yaml files. If no newer version is found no pull request is created. Also, a scheduled cron job will only be triggered on default branches ([see here](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#schedule) for details).

I kept the cron job to be scheduled on every Sunday but maybe it can be increased to every month so that we don't have too many pull requests what do you think ?